lib/new_relic/agent/configuration/default_source.rb

```diff
@@ -2212,6 +2212,73 @@ def self.enforce_fallback(allowed_values: nil, fallback: nil)
           :type => Integer,
           :allowed_from_server => false,
           :description => 'This value represents the total amount of memory available to the host (not the process), in mebibytes (1024 squared or 1,048,576 bytes).'
+        },
+        # security agent
+        :'security.agent.enabled' => {
+          :default => false,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, the security agent is loaded'
+        },
+        :'security.enabled' => {
+          :default => false,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, the security agent is activated'
+        },
+        :'security.mode' => {
+          :default => 'IAST',
+          :external => true,
+          :public => true,
+          :type => String,
+          :allowed_from_server => true,
+          :description => 'Defines the mode for the security agent to operate in. Currently only `IAST` is supported',
+          :dynamic_name => true
+        },
+        :'security.validator_service_url' => {
+          :default => 'wss://csec.nr-data.net',
+          :external => true,
+          :public => true,
+          :type => String,
+          :allowed_from_server => true,
+          :description => 'Defines the end point URL for posting security related data',
+          :dynamic_name => true
+        },
+        :'security.detection.rci.enabled' => {
+          :default => true,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, enables rci detection'
+        },
+        :'security.detection.rxss.enabled' => {
+          :default => true,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, enables rxss detection'
+        },
+        :'security.detection.deserialization.enabled' => {
+          :default => true,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, enables deserialization detection'
+        },
+        :'security.applicationinfo.port' => {
```

Comment:
```suggestion
        :'security.application_info.port' => {
```

Reply:
I can't change this one without a matching merged PR on the security agent itself.

Reply:
Would it be appropriate for me to create a PR for the security agent requesting these changes be made? Or should I start with a Jira ticket?

Reply:
I think Jira would be better as this option is universal and not Ruby specific, so it'll likely need to get circulated amongst those who look at Jira. I'm curious as to how frozen the values are at this point, but I support the change.

Reply:
Whoops - I apologize - the port param IS Ruby specific as it was only needed to be user facing for Ruby. I'll leave it up to you on which approach to take to get it changed, but we can definitely do so without anything impacting the other language agents.

Reply:
Internal Jira ticket requesting changes: https://issues.newrelic.com/browse/NEWRELIC-8926